src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -391,6 +391,25 @@ private async Task<EntryComponent> ExecuteRequests(Hl7.Fhir.Model.Bundle respons
 
                         await request.Handler.Invoke(httpContext);
 
+                        // we will retry a 429 one time per request in the bundle
+                        if (httpContext.Response.StatusCode == (int)HttpStatusCode.TooManyRequests
+                            && _bundleType == BundleType.Batch)
```

Comment:
Why do we care if it is a Batch request? Couldn't we do the same thing for a Transaction request?